src/plugins/rhbz.c

```diff
@@ -941,3 +941,69 @@ xmlrpc_value *rhbz_get_sub_components(struct abrt_xmlrpc *ax, const char *produc
 
     return sub_components;
 }
+
+char *rhbz_get_default_sub_component(const char *component, xmlrpc_value *sub_components)
+{
+    /* Here are some reasonably-looking default subcomponents for components
+     * that have those as of this writing. See if we have a hardcoded deafult,
+     * and check if it actually exists in the sub_components array. Otherwise
+     * just pick whatever comes last.
+     */
+
+    const char *defaults[][2] =
+    {
+            {"binutils",      "system-version"},
+            {"Documentation", "default"},
+            {"dwz",           "system-version"},
+            {"dynist",        "system-version"},
+            {"elfutils",      "system-version"},
+            {"gcc",           "system-version"},
+            {"gdb",           "system-version"},
+            {"kernel",        "Other"},
+            {"kernel-rt",     "Other"},
+            {"kpatch",        "kpatch-utility"},
+            {"ltrace",        "system-version"},
+            {"lvm2",          "Default / Unclassified"},
+            {"make",          "system-version"},
+            {"systemtap",     "system-version"},
+            {"test",          "sub1"},
+            {"valgrind",      "system-version"},
+            {"virtio-win",    "distribution"}
+    };
+    char *candidate = NULL;
+
+    for (int i = 0; i < G_N_ELEMENTS(defaults); i++)
+        if (strcmp(component, defaults[i][0]) == 0)
+        {
+            candidate = strdup(defaults[i][1]);
```

Comment:
Why the `strdup()`?

Reply:
I don't remember any more. Probably just a leftover from an earlier working version of the code. Removed.